projects/models.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  pkgs,
+  sources,
+}:
+let
+  yants = import sources.yants { };
+
+  inherit (yants)
+    string
+    list
+    option
+    attrs
+    enum
+    either
+    ;
+in
+rec {
+  project =
+    p: with p; {
+      name = string name;
+      metadata = with metadata; {
+        summary = string summary;
+        websites = attrs (either (option string) (list string)) {
```

Comment:
In https://github.com/ngi-nix/ngipkgs/issues/504#issuecomment-2666515871, it was metioned that 

> `websites` will now be denormalised because those actually come from each subgrant -- if we keep them there, we can re-aggregate them in presentation if needed.

Does it make sense to look at this PR's model as the representation?